kqueen/auth/common.py

```diff
@@ -140,11 +140,12 @@ def identity(payload):
 
 
 def encrypt_password(_password):
-    config = current_config()
-    rounds = config.get('BCRYPT_ROUNDS', 12)
-    password = str(_password).encode('utf-8')
-    encrypted = bcrypt.hashpw(password, bcrypt.gensalt(rounds)).decode('utf-8')
-    return encrypted
+    if _password:
```

Comment:
why we need this?
seems previous case failed with valid traceback in case of empty pass

Reply:
encrypted successfully generated in case of None

Reply:
and you still get encrypted password for None

Reply:
maybe its better to deny None-passwords at all?
i mean like that
```+++ b/kqueen/blueprints/api/views.py
@@ -512,7 +512,7 @@ def user_password_update(pk):
     if not isinstance(data, dict):
         abort(400)

-    obj.password = encrypt_password(data.get('password'))
+    obj.password = encrypt_password(data['password'])

     try:
         obj.save()```

Reply:
your fix is good, but seems problem is deeper

Reply:
you'll get key error and request will not finish as expected, now 400 - bad request is returned during modal validation - password is None but it is required in Model

Reply:
what about that?
```
diff --git a/kqueen/blueprints/api/views.py b/kqueen/blueprints/api/views.py
index cc63da4..69194a4 100644
--- a/kqueen/blueprints/api/views.py
+++ b/kqueen/blueprints/api/views.py
@@ -512,6 +512,9 @@ def user_password_update(pk):
     if not isinstance(data, dict):
         abort(400)

+    if data.get('password') is None:
+        abort(400, description='Password required')
+
     obj.password = encrypt_password(data.get('password'))

     try:
(END)
```